packages/dds/task-manager/src/taskManager.ts

```diff
@@ -363,6 +375,13 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
             return true;
         }
 
+        if (!this.isAttached()) {
+            // Simulate auto-ack in detached scenario
+            assert(this.runtime.clientId !== undefined, "clientId is undefined");
```

Comment:
This should be valid assertion. Detached mode implies you never had clientId, and connection logic (connection to delta manager) plays out after attach has been completed. I’m curious of any considerations if you are attached, but still not “connected”.? “Connected” here means delta manager connection established (so you have client Id) and your own join op received.

Reply:
> Detached mode implies you never had clientId, and connection logic (connection to delta manager) plays out after attach has been completed.

After a bit more testing, this assert is causing issues. Like you mentioned, the clientId will not be established yet, so this assert will always fail. I'm converting this PR to a draft until this is resolved.

>I’m curious of any considerations if you are attached, but still not “connected”.? “Connected” here means delta manager connection established (so you have client Id) and your own join op received.

Yes, this scenario is already handled before this PR. We will not allow ops to be sent (and throw an error) if you are in a disconnected and attached state

Reply:
This is actually an interesting problem that I hadn't thought about.  Sometimes you will have a client ID (i.e. creating a new TaskManager and attaching it to an already-attached container that you are already connected to), sometimes you won't have a client ID (i.e. creating a new TaskManager for a detached container).

In the first case, it would be nice to actually include your client ID in the attach summary, so that you can continue holding the task through the attach.  In the latter, you need to abandon the tasks during the attach (because at that point you don't know who might get the task before you connect).

LMK if you'd like to chat about this a bit.

Reply:
Sorry, I meant to say NOT be valid. I made two contradicting statements. In short: In detached container you would not have client ID. Matt brought up another interesting point though around attached states (DDs vs container)